src/java/org/apache/cassandra/hints/HintsService.java

```diff
@@ -268,6 +269,31 @@ public synchronized void shutdownBlocking() throws ExecutionException, Interrupt
         bufferPool.close();
     }
 
+    /**
+     * Returns all pending hints that this node has.
+     *
+     * @return a list of {@link PendingHintsInfo}
+     */
+    public List<PendingHintsInfo> getPendingHintsInfo()
+    {
+        return catalog.stores()
+                      .filter(HintsStore::hasFiles)
+                      .map(HintsStore::getPendingHintsInfo)
+                      .collect(Collectors.toList());
+    }
+
+    /**
+     * Returns all pending hints that this node has.
+     *
+     * @return a list of maps with endpoints' ids, total number of hint files, their oldest and newest timestamps.
+     */
+    public List<Map<String, String>> getPendingHints()
```

Comment:
not: I'd probably move `PendingHintsInfo.asMap()` logic here because it is not really supposed to be used somewhere else.

Reply:
I placed `asMap()` inside `PendingHintsInfo` because the nodetool command has to access all keys inside that map, so it is convenient to have these keys available as static constants. Check out

https://github.com/apache/cassandra/blob/af9e91019775d70871f112e2807b91bec44c39d5/src/java/org/apache/cassandra/hints/PendingHintsInfo.java#L52-L55

https://github.com/apache/cassandra/blob/af9e91019775d70871f112e2807b91bec44c39d5/src/java/org/apache/cassandra/tools/nodetool/ListPendingHints.java#L60-L67

It feels to me that having these constants in the `HintsService` class would pollute the namespace. WDYT?

Reply:
Ok, sounds reasonable to me.